vendor/guava-20_0/build.gradle

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * License); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+apply plugin: org.apache.beam.gradle.VendorJavaPlugin
+
+description = "Apache Beam :: Vendored Dependencies :: Guava 20"
+
+vendorJava {
+  dependency = "com.google.guava:guava:20.0"
+  packagesToRelocate = ["com.google.common", "com.google.thirdparty"]
+  intoPackage = "org.apache.beam.guava.v20_0"
+  groupId = "org.apache.beam"
+  artifactId = "beam-vendored-guava-20_0"
+  version = "0.1"
+}
+
+doVendorJava()
```

Comment:
One pitfall is that I had to do something like this. Basically my understanding is that the `VendorJavaPlugin.apply` method runs in the configuration phase but the actual logic we want has to run after _all_ the configuration is complete, so the `VendorJavaExtension` object has its fields written. In all the examples I found, the extension object is only accessed via tasks and other things that happen at a later phase. I think that basically plugins causing configuration seems very badly supported.